hibernate-core/src/main/java/org/hibernate/query/criteria/internal/predicate/InPredicate.java

```diff
@@ -197,14 +201,51 @@ public String render(boolean isNegated, RenderingContext renderingContext) {
 				}
 			}
 			else {
+				Expression lastValue = null;
 				buffer.append( '(' );
 				String sep = "";
 				for ( Expression value : values) {
+					lastValue = value;
```

Comment:
I think we don't need to assign it multiple times (suppose our list has 100 items, we need to assign to `lastValue` 100 times). We can simply get its value and assign it only once outside of the loop.
```